sdk/core/azure-core/src/http/body_stream.cpp

```diff
@@ -53,7 +54,7 @@ std::vector<uint8_t> BodyStream::ReadToEnd(Context const& context, BodyStream& b
 
   for (auto chunkNumber = 0;; chunkNumber++)
   {
-    buffer.resize((chunkNumber + 1) * chunkSize);
+    buffer.resize((static_cast<decltype(buffer)::size_type>(chunkNumber) + 1) * chunkSize);
```

Comment:
Why is this necessary and what is this cast doing exactly?

Reply:
We remove a warning, stating our intentions more clearly. `resize()` takes `vector<uint8_t>::size_type`, which is defined by implementation, but realistically is `uint64`. `chunkNumber` is `int`.